### PR TITLE
Add DataFrame/Series.align.

### PR DESCRIPTION
**Pull request by ueshin**
Add `DataFrame/Series.align` with parameters `join`, `axis`, and `copy` supported.

- https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.align.html
- https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.align.html

### CODE REVIEW

File: databricks/koalas/utils.py [hunk 1]
```diff
@@ -667,6 +668,25 @@ def validate_bool_kwarg(value, arg_name):
     return value
 
 
+def validate_how(how: str) -> str:
```
Comment (itholic) [inline databricks/koalas/utils.py:671]: +1

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -10916,6 +10905,180 @@ def tail(self, n=5) -> "DataFrame":
...
+
+        how = validate_how(join)
+        axis = validate_axis(axis, None)
+
+        right_is_series = isinstance(other, Series)
+        if right_is_series:
+            if axis is None:
+                raise ValueError("Must specify axis=0")
+            elif axis != 0:
+                raise NotImplementedError("align currently only works for axis=0")
```
Comment (itholic) [inline databricks/koalas/frame.py:11035]: nit: "align currently only works for axis=0 when right is Series" ??

**Review by itholic — COMMENTED** [review]
Seems good to me

File: databricks/koalas/utils.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/utils.py:671]: So clear!

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -10916,6 +10905,180 @@ def tail(self, n=5) -> "DataFrame":
...
+        if not isinstance(other, (DataFrame, Series)):
+            raise TypeError("unsupported type: {}".format(type(other).__name__))
+
+        how = validate_how(join)
+        axis = validate_axis(axis, None)
+
+        right_is_series = isinstance(other, Series)
+        if right_is_series:
+            if axis is None:
+                raise ValueError("Must specify axis=0")
```
Comment (xinrong-databricks) [inline databricks/koalas/frame.py:11033]: I'm afraid we might forget to append `or 1` to the end of the error message when `axis=1` is supported.

File: databricks/koalas/frame.py [hunk 4]
```diff
@@ -10916,6 +10905,180 @@ def tail(self, n=5) -> "DataFrame":
...
+
+            if right_is_series:
+                right = first_series(right).rename(other.name)
+
+        if (
+            axis is None or axis == 1
+        ) and left._internal.column_labels != right._internal.column_labels:
+
+            if left._internal.column_labels_level != right._internal.column_labels_level:
+                raise ValueError("cannot join with no overlapping index names")
```
Comment (xinrong-databricks) [inline databricks/koalas/frame.py:11053]: Wouldn't the error message complain about a different level of the column index?

**Comment by xinrong-databricks** [comment]
Left some questions, otherwise LGTM!

File: databricks/koalas/frame.py (hunk 4 above)
Comment (ueshin) [reply databricks/koalas/frame.py:11055]: The error message follows pandas' one:

```py
>>> pdf1 = pd.DataFrame({"a": [1, 2, 3], "b": ["a", "b", "c"]}, index=[10, 20, 30])
>>> pdf2 = pd.DataFrame({("x", "a"): [4, 5, 6], ("y", "c"): ["d", "e", "f"]}, index=[10, 11, 12])
>>>
>>> pdf1.align(pdf2)
Traceback (most recent call last):
...
ValueError: cannot join with no overlapping index names
```

Comment (xinrong-databricks) [reply databricks/koalas/frame.py:11055]: Makes sense!

**Review by itholic — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**